node/node-core/src/eth/trie.rs

```diff
@@ -0,0 +1,46 @@
+//! Utility functions for Ethereum adapted from https://github.dev/rust-blockchain/ethereum/blob/755dffaa4903fbec1269f50cde9863cf86269a14/src/util.rs
```

Comment:
Is it possible that Akula's trie module has all we need? https://github.com/akula-bft/akula/tree/master/src/trie

In that case, it might be worth PR'ing a refactor of that module in a separate crate and we can use that?

Maybe that's wrong, so feel free to ignore if so

Reply:
hard to say, couldn't find any examples there so I'd rather stick with that for now